File: sentimental-credit/credit.py
import cs50


def main():
    credit = cs50.get_string("Number: ")
    valid_sum = check_sum(credit)
    length = len(credit)
    first_digits = int(credit[0:2])

    if valid_sum:
        if length == 15 and (first_digits == 34 or first_digits == 37):
            print("AMEX")
        elif length == 16 and first_digits >= 51 and first_digits <= 55:
            print("MASTERCARD")
        elif length == 16 or length == 13 and first_digits / 10 == 4:
            print("VISA")
        else:
            print("INVALID")
    else:
        print("INVALID")


def check_sum(credit):
    i = sum = 0
    for digit in credit[::-1]:
        if i % 2 == 0:
            sum += int(digit)
        else:
            digit = str(int(digit) * 2)
            for d in digit:
                sum += int(d)
        i += 1
    return True if sum % 10 == 0 else False


main()

File: credit/credit.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

bool checkSum(long credit)
{
    int digit, i, sum;
    i = sum = 0;
    while (credit)
    {
        digit = credit % 10;
        if (i % 2 == 0)
        {
            sum += digit;
        }
        else
        {
            digit *= 2;
            while (digit)
            {
                sum += digit % 10;
                digit /= 10;
            }
        }
        credit /= 10;
        i++;
    }
    printf("SOMA: %i\n", sum);
    bool sumcheck = sum % 10 == 0 ? true : false;
    return sumcheck;
}

int digitsCounter(long credit)
{
    int digits = 0;
    while (credit)
    {
        credit /= 10;
        digits++;
    }
    return digits;
}

int firstDigits(long credit, int digits)
{
    return credit / pow(10, (digits - 2));
}

int main(void)
{
    long credit;
    // gets card number
    do
    {
        credit = get_long("Number: ");
    }
    while (!credit);
    // does checksum
    bool checksum = checkSum(credit);
    // counts the digits
    int digits = digitsCounter(credit);
    // gets the first two digits
    int first_digits = firstDigits(credit, digits);

    printf("%i\n", digits);
    printf("%i\n", first_digits);
    printf("%i\n", checksum);

    if (checksum)
    {
        // checks if it's AMEX
        if (digits == 15 && (first_digits == 34 || first_digits == 37))
        {
            printf("AMEX\n");
        }
        // checks if it's MASTERCARD
        else if (digits == 16 && (first_digits >= 51 && first_digits <= 55))
        {
            printf("MASTERCARD\n");
        }
        // checks if it's VISA
        else if ((digits == 16 || digits == 13) && (first_digits / 10 == 4))
        {
            printf("VISA\n");
        }
        else
        {
            printf("INVALID\n");
        }
    }
    else
    {
        printf("INVALID\n");
    }



}